Feature/update next event

athackst opened the pull request.

athackst committed:
convert to general use

athackst committed:
added info for debugging

athackst committed:
update app to not require worker provision

athackst committed:
fix formatting, update readme

athackst committed:
add dependabot config

athackst committed:
remove changelong

athackst committed:
added info and info spec

athackst committed:
update readme

athackst committed:
add deploy action when updates pushed to main

athackst committed:
update actions to use bundler-cache

athackst committed:
fix typo

athackst committed:
update next event slack message

MikeRogers0 commented:
@athackst this is so cool!!! Am I OK to pull a few of your changes into this repo? It looks like you've improved it a bunch :D

athackst commented:
Thanks!

I was going to open an issue to see if you all would want a PR.  I forgot that PRs default to base repos and accidentally opened this before I was ready (whoops!) :facepalm: 

My group also makes use of Meetingplace, so I used your excellent start to generalize it enough for our needs.

In particular:
1. I added a group info task that returns the description of the Meetingplace group.  I mostly did this just to be able to test that I'd integrated with Slack right without having to have an event within a certain time frame.
2. I added the ability to define channel and group with each task so that I could hook different channels up to different groups.  We're a collection of chapters and each chapter has its own Meetingplace group.  This is optional and otherwise uses the environment configuration for a single group->channel combo if not set.
3. I added deploy to Heroku button so other groups could spin up their own dynos.
4. I added some developer things -- dev container for VSCode, tasks, workflow to CD to Heroku, Dependabot

I wasn't sure what kind of changes you'd be interested in consuming
1. Do you want to completely subsume this and have this repo be a general Meetingplace bot?
2. Do you want to reference my general fork as a gem you can build custom virtual-coffee needs against?  I could either publish this as a gem or you could [install from the git repo](https://bundler.io/guides/git.html)
3. Do you want to just merge what I have now and split it off from there?
4. Do you want just a subset of my changes?

Then I can open the relevant PR :smile:

MikeRogers0 commented:
> Do you want just a subset of my changes?

I think for now I'd just like the changes you made to the `.github` files :) I'll cherry pick your commits into a new PR if you'd like? :)

My current plans for this repo is to keep the virtual coffee bot kind of specific to virtual coffee (I'm planning to make it do a bit more then meetingplace). Though I totally support what you're doing & I'm happy to help out if you'd like :)

athackst commented:
Sure! SGTM  

Feel free to take whatever you find useful. Let me know if you have any questions. 😄
